### DISCUSSION TITLE
en_US LJSpeech FastSpeech + MB Melgan Training Quality Question

### DISCUSSION THREAD

**Discussion by iprovalo** [coqui-ai/TTS#1667] [General Q&A] [answered]
I am training on en_US LJ Speech Dataset, both FastSpeech and MB Melgan.  While testing the wav file comes out with just some noise after about 1m steps.

Here is the short system description:

```
        "PyTorch_version": "1.11.0+cu115",
        "TTS": "0.6.2",
        "numpy": "1.21.6"
        "python": "3.7.13",
```

1. I merged the ljspeech dataset from multiple folders (gender, book, speaker) into a single folder. 

2. Commands for Fast Speech:
`python3 /apps/tts/TTS/recipes/ljspeech/fast_speech/train_fast_speech_en.py`
Where `BaseDatasetConfig` has an override: `language='en',` and `BaseAudioConfig` has this override: `sample_rate=16000,`

3. MB Melgan:
`python3 /apps/tts/TTS/recipes/ljspeech/multiband_melgan/train_multiband_melgan_en.py`

4. Running inference by pointing to the running job:
```tts --text "Let's test this sentence." --model_path /apps/tts/data/output/coquiai/en_US/fast_speech_ljspeech-June-09-2022_03+00PM-0000000/best_model.pth --config_path /apps/tts/data/output/coquiai/en_US/fast_speech_ljspeech-June-09-2022_03+00PM-0000000/config.json --out_path /apps/tts/data/output/coquiai/speech_en.wav --vocoder_path /apps/tts/data/output/coquiai/en_US/run-June-09-2022_05+40PM-0000000/best_model.pth --vocoder_config_path /apps/tts/data/output/coquiai/en_US/run-June-09-2022_05+40PM-0000000/config.json```

5. Here are the EvalStats average loss function for Fast Speech:
![Screen Shot 2022-06-18 at 12 36 29 PM](https://user-images.githubusercontent.com/1660997/174455213-22845be8-1ebc-4a79-9a0d-adcb35d04808.png)

6. MB Melgan EvalStats:
![Screen Shot 2022-06-18 at 12 36 34 PM](https://user-images.githubusercontent.com/1660997/174455235-87612a84-239a-435b-a6cd-6ad86e897f94.png)

Am I missing some steps?

**Comment by iprovalo** [comment]
I started over with the 13K ljspeech listed in the [tutorial](https://tts.readthedocs.io/en/latest/tutorial_for_nervous_beginners.html). 

After 30K steps, here is the loss:

![Screen Shot 2022-06-21 at 11 40 45 AM](https://user-images.githubusercontent.com/1660997/174877267-7cf25abd-b319-4ff9-9184-e750e15df7f5.png)

...the audio is somewhat robot-sounding, but it is coming through with this command:

`tts --text "This is a demo" --model_path /apps/tts/data2/output/run-June-21-2022_05+29AM-71281ff1/best_model.pth  --config_path /apps/tts/data2/output/run-June-21-2022_05+29AM-71281ff1/config.json`

Here I am using my own trained `glow_tts` from the tutorial, and a default Griffin-Lim vocoder:
`Griffin-Lim as no vocoder model defined`

I will be training the vocoder next on the same dataset.

I think the issue is with the sample rate in the ljspeech dataset I used in the original run (16000 Hz): 
https://www.caito.de/2019/01/03/the-m-ailabs-speech-dataset/
https://data.solak.de/data/Training/stt_tts/en_US.tgz

vs 

Tutorial LJSpeech dataset (22050 Hz): 
https://keithito.com/LJ-Speech-Dataset/
https://data.keithito.com/data/speech/LJSpeech-1.1.tar.bz2

I am still curious why is it that in both cases (my original run and this tutorial-based run), on the tensorboard an audio file would be a somewhat recognizable speech during the training process, but when I ran the inference, it would be just noise for my original run case.

**Comment by iprovalo** [comment, marked as answer]
In addition to this sample rate mismatch, I was using the `ljspeech` dataset in a single speaker way, i.e. merging multiple speaker `wavs` into a single folder.  @WeberJulian helped me with a hint to use a different dataset type which would use the folder structure to assign the speakers automatically:  

`BaseDatasetConfig(name="mailabs"...)`

Thanks, @WeberJulian!

I think I will need the following as well:
`    use_speaker_embedding=True,`

and this from `vctk` example:

```
speaker_manager = SpeakerManager()
speaker_manager.set_ids_from_data(train_samples + eval_samples, parse_key="speaker_name")
config.num_speakers = speaker_manager.num_speakers
```

**Comment by erogol** [comment]
Thanks for the detailed progress report here. I think it is helpful for people going through the same challenge.
